index.js

```diff
@@ -0,0 +1,42 @@
+module.exports = function(db, credentials, initialRefreshToken) {
+const knex = require('knex')(db),
+      SimpleOAuth2 = require('simple-oauth2'),
+      oauth2 = SimpleOAuth2.create(credentials);
+
+  var migrated;
+  function migrate() {
+    if (migrated) return migrated;
+    migrated = knex.migrate.latest({ directory: __dirname + '/migrations' });
+    return migrated;
+  }
+
+  function latestToken(type) {
+    return migrate().then(function() {
+      return knex(type + '_tokens').orderBy('id', 'desc').first(type + '_token').then(function(result) {
```

Comment:
not a big fan of ordering by `id` even if it is auto incrementing.  Would be better to use `created_at` or `updated_at` values instead.